src/pages/PoolsPage/v3/SupplyLiquidityV3/components/RangeSelector/index.tsx

```diff
@@ -0,0 +1,322 @@
+  const usdPriceA = useUSDCPrice(tokenA ?? undefined);
+  const usdPriceB = useUSDCPrice(tokenB ?? undefined);
+
+  const initialUSDPrices = useInitialUSDPrices();
+  const initialTokenPrice = useInitialTokenPrice();
+
+  const handleOnBlur = useCallback(() => {
+    if (isUSD && usdPriceB) {
+      if (tokenB?.wrapped.address === USDC.address) {
+        onUserInput(localUSDValue);
+      } else {
+        if (tokenValue && tokenValue.trade) {
+          onUserInput(
+            String(
+              +localUSDValue * +tokenValue.trade?.outputAmount.toSignificant(5),
+            ),
+          );
+          setLocalTokenValue(
+            String(+localUSDValue * +usdPriceB.toSignificant(5)),
+          );
+        } else {
+          onUserInput(localUSDValue);
+          setLocalTokenValue(localUSDValue);
+        }
+      }
+    } else if (isUSD && initialUSDPrices.CURRENCY_B) {
+      if (tokenB?.wrapped.address === USDC.address) {
+        onUserInput(localUSDValue);
+      } else {
+        onUserInput(String(+localUSDValue / +initialUSDPrices.CURRENCY_B));
+        setLocalTokenValue(
+          String(+localUSDValue / +initialUSDPrices.CURRENCY_B),
+        );
+      }
+    } else if (isUSD && initialTokenPrice && usdPriceA) {
+      if (tokenB?.wrapped.address === USDC.address) {
+        onUserInput(localUSDValue);
+      } else {
+        onUserInput(
+          String(
+            +localUSDValue * +initialTokenPrice * +usdPriceA.toSignificant(5),
+          ),
+        );
+        setLocalTokenValue(
+          String(
+            +localUSDValue * +initialTokenPrice * +usdPriceA.toSignificant(5),
+          ),
+        );
+      }
+    } else if (!isUSD) {
+      if (usdPriceB) {
+        setLocalUSDValue(
+          String(+localTokenValue * +usdPriceB.toSignificant(5)),
+        );
+      } else if (initialUSDPrices.CURRENCY_B) {
+        setLocalUSDValue(
+          String(+localTokenValue * +initialUSDPrices.CURRENCY_B),
+        );
+      }
+      onUserInput(localTokenValue);
+    }
+  }, [
+    localTokenValue,
+    localUSDValue,
+    tokenValue,
+    valueUSD,
+    usdPriceB,
+    onUserInput,
+  ]);
+
+  // for button clicks
+  const handleDecrement = useCallback(() => {
+    onUserInput(decrement());
+  }, [decrement, onUserInput]);
+
+  const handleIncrement = useCallback(() => {
+    onUserInput(increment());
+  }, [increment, onUserInput]);
+
+  useEffect(() => {
+    if (value) {
+      setLocalTokenValue(value);
+      if (value === '∞') {
+        setLocalUSDValue(value);
+        return;
+      }
+      if (usdPriceB) {
+        setLocalUSDValue(String(+value * +usdPriceB.toSignificant(5)));
+      } else if (initialUSDPrices.CURRENCY_B) {
+        setLocalUSDValue(String(+value * +initialUSDPrices.CURRENCY_B));
+      } else if (initialTokenPrice && usdPriceA) {
+        setLocalUSDValue(
+          String(+value * +initialTokenPrice * +usdPriceA.toSignificant(5)),
```

Comment:
in the future the toSignificant should be bound by the token decimals e.g Tel coin has 2 decimals it doesn't make sense to use decimals beyond that position.  This applies to the whole code based not just here